asr1k_neutron_l3/models/neutron/l3/access_list.py

```diff
@@ -20,7 +20,10 @@
 
 class AccessList(base.Base):
     def __init__(self, id, drop_on_17_3=False):
-        super(AccessList, self).__init__()
+        # If we use super(AccessList, self) here the MRO of type(self) is determining the __init__
+        # we call. If self is a child, that could lead to not calling base.Base.__init__().
+        # If the second argument to super is a type, the returned method is unbound, hence we pass self to it.
+        super(AccessList, AccessList).__init__(self)
```

Comment:
Jep, the problem comes from  [`firewall.AccessList`](https://github.com/sapcc/asr1k-neutron-l3/blob/fwaas-support/asr1k_neutron_l3/models/neutron/l3/firewall.py#L53) which inherits from `FirewallPolicyObject` but also from this class.  `FirewallPolicyObject` in turn also  inherits from `base.Base` again.  So a diamond shaped inheritance.
Say `firewall.AccessList.__init__()` calls to `access_list.AccessList.__init__()`, now I want to prevent that the `super()` in line 26 calls `FirewallPolicyObject.__init__()` but instead `base.Base().__init__()`

I am very open to other suggestions here.